src/view/xwayland.cpp

```diff
@@ -326,6 +327,16 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
 
         on_request_move.set_callback([&] (void*) { move_request(); });
         on_request_resize.set_callback([&] (void*) { resize_request(); });
+        on_request_activate.set_callback([&] (void*)
+        {
+            if (!this->activated)
+            {
+                view_self_request_focus_signal data;
+                data.view = this;
+                this->get_output()->emit_signal("view-self-request-focus", &data);
```

Comment:
In theory it is possible to have `this->get_output()` return `NULL`. The codebase is not really ready for that but we should check it for new code (also same thing for the other signal). Aside from that LGTM.